WordPress/gutenberg
# Add text domain option while scaffolding the block in create-block

## What?


Fixes #54980 

## Why?


- This PR adds a new option prompt in the `create-block` command to add `text domain` option in the scaffolding block's block.json metadata

## How?


- As per the flow of the command goes:
 - If the user calls `npx @wordpress/create-block --no-plugin` it adds 'textdomain' prompts where the user can specify the text domain for that block.
 - Created a new prompt in the prompts.js for textdomain and added it to scaffold.js and in the plugin template for `--no-plugin` option. This way it will be shown both with the plugin and no plugin option.

## Testing Instructions





- Open the code editor and terminal
- Create a folder inside `wp-content/plugins`
- Run `npx ../gutenberg/packages/create-block --no-plugin`. (Make sure you already have the Gutenberg plugin installed and updated with this branch code in plugins folder already)
- You will see an option `The text domain used to internationalize text in the block:` after selecting a category where you can specify your text-domain.
- After the block scaffolding is done, you will be able to see your specified text-domain in `block.json` of your block folder




### Testing Instructions for Keyboard

None

## Screenshots or screencast 
<img width="765" alt="Screenshot 2023-12-19 at 2 58 54 PM" src="https://github.com/WordPress/gutenberg/assets/55312427/8da9ad12-00c9-4192-b7d6-07887ccfa376">
<img width="777" alt="Screenshot 2023-12-19 at 2 59 09 PM" src="https://github.com/WordPress/gutenberg/assets/55312427/197aef85-ca33-4692-98a9-f4bd447d1482">
